src/Symfony/Component/TypeInfo/TypeContext/TypeContextFactory.php

```diff
@@ -113,6 +108,56 @@ public function createFromReflection(\Reflector $reflection): ?TypeContext
         );
     }
 
+    private function createNewInstanceFromClassName(
+        string $calledClassName,
+        string $declaringClassName,
+    ): TypeContext {
+        $calledClassPath = explode('\\', $calledClassName);
+        $declaringClassPath = explode('\\', $declaringClassName);
+
+        $declaringClassReflection = self::$reflectionClassCache[$declaringClassName] ??= new \ReflectionClass($declaringClassName);
+
+        $typeContext = $this->getIntermediateTypeContext(
+            end($calledClassPath),
+            $declaringClassReflection,
+        );
+
+        return new TypeContext(
+            $typeContext->calledClassName,
+            $typeContext->declaringClassName,
+            $typeContext->namespace,
+            $typeContext->uses,
+            $this->collectTemplates($declaringClassReflection, $typeContext),
+            $this->collectTypeAliases($declaringClassReflection, $typeContext),
+        );
+    }
+
+    private function getIntermediateTypeContext(
+        string $calledClassShortName,
+        \ReflectionClass $declaringClassReflection,
+    ): TypeContext {
```

Comment:
our CS is to always have function signatures on one line (except for constructor using CPP)
```suggestion
    ): TypeContext
    {
```

Reply:
fixed